web/src/timestamp_width_handler.js

```diff
@@ -0,0 +1,231 @@
+    "pt-PT": {
+        maxtimestamp12_hr: "11:59 da tarde",
+        maxtimestamp24_hr: "23:59",
+        name: "Portuguese (Portugal)",
+    },
+    ro: {
+        maxtimestamp12_hr: "11:59 p.m.",
+        maxtimestamp24_hr: "23:59",
+        name: "Română",
+    },
+    si: {
+        maxtimestamp12_hr: "ප.ව. 11:59",
+        maxtimestamp24_hr: "23:59",
+        name: "Sinhala",
+    },
+    fi: {
+        maxtimestamp12_hr: "11:59 ip.",
+        maxtimestamp24_hr: "23:59",
+        name: "Suomi",
+    },
+    sv: {
+        maxtimestamp12_hr: "11:59 em",
+        maxtimestamp24_hr: "23:59",
+        name: "Svenska",
+    },
+    tl: {
+        maxtimestamp12_hr: "11:59 PM",
+        maxtimestamp24_hr: "23:59",
+        name: "Tagalog",
+    },
+    vi: {
+        maxtimestamp12_hr: "11:59 CH",
+        maxtimestamp24_hr: "23:59",
+        name: "Tiếng Việt",
+    },
+    tr: {
+        maxtimestamp12_hr: "ÖS 11:59",
+        maxtimestamp24_hr: "23:59",
+        name: "Türkçe",
+    },
+    be: {
+        maxtimestamp12_hr: "11:59 PM",
+        maxtimestamp24_hr: "23:59",
+        name: "Беларуская",
+    },
+    bg: {
+        maxtimestamp12_hr: "11:59 ч. сл.об.",
+        maxtimestamp24_hr: "23:59 ч.",
+        name: "Български",
+    },
+    ru: {
+        maxtimestamp12_hr: "11:59 PM",
+        maxtimestamp24_hr: "23:59",
+        name: "Русский",
+    },
+    sr: {
+        maxtimestamp12_hr: "11:59 PM",
+        maxtimestamp24_hr: "23:59",
+        name: "Српски",
+    },
+    uk: {
+        maxtimestamp12_hr: "11:59 пп",
+        maxtimestamp24_hr: "23:59",
+        name: "Українська",
+    },
+    ar: {
+        maxtimestamp12_hr: "١١:٥٩ م",
+        maxtimestamp24_hr: "٢٣:٥٩",
+        name: "العربيّة",
+    },
+    fa: {
+        maxtimestamp12_hr: "۱۱:۵۹ بعدازظهر",
+        maxtimestamp24_hr: "۲۳:۵۹",
+        name: "فارسی",
+    },
+    hi: {
+        maxtimestamp12_hr: "11:59 PM",
+        maxtimestamp24_hr: "23:59",
+        name: "हिंदी",
+    },
+    ta: {
+        maxtimestamp12_hr: "பிற்பகல் 11:59",
+        maxtimestamp24_hr: "23:59",
+        name: "தமிழ்",
+    },
+    ml: {
+        maxtimestamp12_hr: "11:59 PM",
+        maxtimestamp24_hr: "23:59",
+        name: "മലയാളം",
+    },
+    ko: {
+        maxtimestamp12_hr: "오후 11:59",
+        maxtimestamp24_hr: "23:59",
+        name: "한국어",
+    },
+    ja: {
+        maxtimestamp12_hr: "午後11:59",
+        maxtimestamp24_hr: "23:59",
+        name: "日本語",
+    },
+    "zh-CN": {
+        maxtimestamp12_hr: "下午11:59",
+        maxtimestamp24_hr: "23:59",
+        name: "简体中文",
+    },
+};
+
+export function calculateFormattedTimeWidth() {
+    const locale = timerender.get_user_locale();
+    const localeData = candidate_timestamps[locale];
+    let formattedTimeWidth = 45; // Default width if locale is not found
+
+    if (localeData) {
+        // timestamp contains the string according to timeformat.
+        const timestamp = user_settings.twenty_four_hour_time
```

Comment:
What should be happening here is passing a date object to probably `timerender.stringify_time()`--study the code so you can see EXACTLY how timestamps are being added to messages. If your code doesn't replicate that 100%, this is not going to address the issue.